Update format.asciidoc to describe strict_date_optional_time_nanos (#57527)
elastic/elasticsearch: docs/reference/mapping/params/format.asciidoc

[[strict-date-time]]`date_optional_time` or `strict_date_optional_time`::

>     A generic ISO datetime parser where the date, with at least a year part is
>     mandatory and the time, separated by `T`, is optional.

Reviewer: ```suggestion
    A generic ISO datetime parser, where the date must include the year at a minimum, and the time
    (separated by `T`), is optional.
```

    Examples: `yyyy-MM-dd'T'HH:mm:ss.SSSZ` or  `yyyy-MM-dd`.

[[strict-date-time-nanos]]`strict_date_optional_time_nanos`::

>     A generic ISO datetime parser where the date, with at least a year part is
>     mandatory and the time, separated by `T`, is optional. Fraction of a second
>     part has a nanosecond resolution

Reviewer: ```suggestion
    A generic ISO datetime parser, where the date must include the year at a minimum, and the time
    (separated by `T`), is optional. The fraction of a second
    part has a nanosecond resolution.
```
